### PR TITLE
Shortest Path between two cells in a Matrix using BFS

### LINKED ISSUE #798: BFS on a matrix in C++, Java and Python
Computing Shortest path between two cells in a matrix

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #798 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
-  Added implementation for shortest path between source and destination cells in a matrix using Breadth First Search

#### Languages Used:
- C++
- Java
- Python

#### Files Added:
- BFS_On_Matrix.cpp
- BFS_On_Matrix.java
- BFS_On_Matrix.py

### CODE REVIEW

File: BFS_On_Matrix/BFS_On_Matrix.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            // if distance to reach the cell to the left is less than the computed previous path distance, update it
+                dist[x][y - 1] = dist[x][y] + 1;
+                q.push(mkp(x, y - 1));
+            }
+        }
+
+        // If move towards right is allowed or it is the destination cell
+        if (y + 1 < col && (mat[x][y + 1] == 1 || mat[x][y + 1] == 3))
+        {
+        // if distance to reach the cell to the right is less than the computed previous path distance, update it
```
Comment (jainaman224) [inline BFS_On_Matrix/BFS_On_Matrix.cpp]: Indentation fix.

File: BFS_On_Matrix/BFS_On_Matrix.cpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                    {1, 0, 1, 1, 1},
+                    {0, 1, 1, 1, 0},
+                    {3, 1, 0, 0, 1},
+                    {1, 1, 0, 0, 1}};
+
+    // Call to function to compute distance between source and destination
+    int answer = findDistance(row, col, mat);
+
+    // When source and destination are unreachable
+    if(answer == INT_MAX)
```
Comment (jainaman224) [inline BFS_On_Matrix/BFS_On_Matrix.cpp]: White space issue.

File: BFS_On_Matrix/BFS_On_Matrix.cpp (hunk 1 above)
Comment (singh-shreya6) [reply BFS_On_Matrix/BFS_On_Matrix.cpp]: fixed

File: BFS_On_Matrix/BFS_On_Matrix.cpp (hunk 2 above)
Comment (singh-shreya6) [reply BFS_On_Matrix/BFS_On_Matrix.cpp]: fixed

**Comment by jainaman224** [comment]
Squash commits and it is ready to be merged.

**Comment by singh-shreya6** [comment]
Done.

**Pull request merged**